libuavcan_drivers/stm32/driver/src/uc_stm32_can.cpp

```diff
@@ -472,6 +477,9 @@ bool CanIface::waitMsrINakBitStateChange(bool target_state)
 #if UAVCAN_STM32_CHIBIOS
         ::chThdSleep(MS2ST(1));
 #endif
+#if UAVCAN_STM32_FREERTOS
+        //::osDelay(1);
```

Comment:
Some sort of rescheduling delay is necessary here. Probably you want to uncomment this call.